README.md

```diff
@@ -101,3 +101,100 @@ The command line interface also gives a quick overview of the different options
    changed by setting the `JUPYTERHUB_REQUEST_TIMEOUT` environment variable.
+
+## How it works
+
+jupyterhub-idle-culler culls user servers using JupyterHub's REST API
+([/users/{name}/server](https://jupyterhub.readthedocs.io/en/stable/_static/rest-api/index.html#operation--users--name--server-delete)
+or
+[/users/{name}/servers/{server_name}](https://jupyterhub.readthedocs.io/en/stable/_static/rest-api/index.html#operation--users--name--servers--server_name--delete)),
+and makes the culling decisions based on its configuration and what JupyterHub
+reports about the user servers via its REST API
+[(/users)](https://jupyterhub.readthedocs.io/en/stable/_static/rest-api/index.html#path--users)
+where user servers' `last_activity` is reported back.
+
+The `last_activity` that JupyterHub reports is the most recent summary of
+information updated at a regular interval via the [`update_last_activity`
+function](https://github.com/jupyterhub/jupyterhub/blob/1.4.2/jupyterhub/app.py#L2646)
+that combines two sources of information.
+
+1. **The proxy's routes data**
+
+   The `update_last_activity` function will [ask the
+   proxy](https://jupyterhub.readthedocs.io/en/stable/reference/proxy.html#retrieving-routes)
+   for the active routes and collects associated `last_activity` data if it is
+   available.
+
+   `last_activity` data for routes will be available when using
+   [configurable-http-proxy](https://github.com/jupyterhub/configurable-http-proxy#readme)
+   as JupyterHub does by default, but if for example
+   [traefik-proxy](https://github.com/jupyterhub/traefik-proxy#readme) is used,
+   no such data will be available.
+
+2. **The user server's activity reports**
+
+   The `update_last_activity` function also reads JupyterHub's database that
+   keeps state about servers `last_activity`. These database records are updated
+   whenever a server notifies JupyterHub about activity, as they are
+   responsibility to do.
+
+   As the servers need to be aware of JupyterHub to notify it about activity,
+   they are supposed to be started via the
+   [`jupyterhub-singleuser`](https://github.com/jupyterhub/jupyterhub/blob/1.4.2/setup.py#L115)
+   script that is made available by installing jupyterhub (or `jupyterhub-base`
+   on conda-forge).
+
+   The `jupyterhub-singleuser` script launches a modified server application
+   that keeps JupyterHub updated with the server activity via the
+   [`notify_activity`](https://github.com/jupyterhub/jupyterhub/blob/1.4.2/jupyterhub/singleuser/mixins.py#L497)
+   function.
+
+   The `notify_activity` function in turn make use of the server applications
+   `last_activity` function (see implementation in
+   [NotebookApp](https://github.com/jupyter/notebook/blob/v6.4.0/notebook/notebookapp.py#L392-L397)
+   and
+   [ServerApp](https://github.com/jupyter-server/jupyter_server/blob/v1.9.0/jupyter_server/serverapp.py#L375)
+   respectively) that that combines information from API activity, kernel
```

Comment:
If you're using additional apps like rstudio with jupyter-server-proxy, activity there is also included

Reply:
Because jupyter-server-proxy registers an [last_activity_times](https://github.com/jupyter-server/jupyter_server/blob/42c27d913348cd91841b85b23f1f6085a4ed5132/jupyter_server/serverapp.py#L290-L292) entry?

Reply:
Nope, via api activity: https://github.com/jupyterhub/jupyter-server-proxy/blob/0f1185f77d2273c0ccac6e0c438bef62b2288695/jupyter_server_proxy/handlers.py#L114-L120.